File: src/Modelo/Trabalho.java
package Modelo;

import java.util.Calendar;
import java.text.SimpleDateFormat;

public class Trabalho extends Atividade{
    
    private static final long serialVersionUID = 1L;
    int numMaxPorGrupo;
    int cargaHoraria;

    public Trabalho(String nome, Boolean sincrona, Calendar prazo, int numMaxPorGrupo, int cargaHoraria){
        super(nome, sincrona);
        this.data = prazo;
        this.numMaxPorGrupo = numMaxPorGrupo;
        this.cargaHoraria = cargaHoraria;
    }

    @Override
    public int obterCargaHoraria(){
        return cargaHoraria;
    }

    public boolean ehAvaliativa(){
        return true;
    }

    @Override
    public String toString(){
        SimpleDateFormat dt1 = new SimpleDateFormat("dd/MM/yyyy");
        return dt1.format(data.getTime());
    }
}

File: cpp_version/Modelo/Trabalho.cpp
#include "Trabalho.h"

Trabalho :: Trabalho (string nome, bool sincrona, time_t prazo, int numMaxPorGrupo, int cargaHoraria) : Atividade (nome, sincrona){
    this->data = prazo;
    this->numMaxPorGrupo = numMaxPorGrupo;
    this->cargaHoraria = cargaHoraria;
}

int Trabalho :: obterCargaHoraria(){
    return cargaHoraria;
}

bool Trabalho :: ehAvaliativa(){
    return true;
}